### PR TITLE
Use Braintree iOS SDK v5

### PR DESCRIPTION
**Pull request by sestevens**
### Summary of changes

 Updates for Braintree iOS SDK v5:
 - Use new PayPal and Venmo methods
 - Change `BTAppSwitch` to `BTAppContextSwitcher`
 - Remove use of `BTAppSwitchDelegate`
 - Update podspec, Cartfile and Package.swift

**TO DO:** Investigate why the CarthageTest app isn't building. We are able to build the demo app and SPMTest, and `pod lib lint` succeeds.

 ### Checklist

 - [x] Added a changelog entry

- @sestevens 
- @hollabaq86

### CODE REVIEW

File: BraintreeDropIn.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -14,7 +14,6 @@
 		03B793AD1DAD61B500F54B1B /* BTDropInResultTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 03B793AC1DAD61B500F54B1B /* BTDropInResultTests.m */; };
 		4225EA53257EC1EB00923CE3 /* BTPaymentMethodNonce+DropIn.h in Headers */ = {isa = PBXBuildFile; fileRef = 4225EA52257EC1EB00923CE3 /* BTPaymentMethodNonce+DropIn.h */; settings = {ATTRIBUTES = (Private, ); }; };
 		4225EA58257EC36300923CE3 /* BTPaymentMethodNonce+DropIn.m in Sources */ = {isa = PBXBuildFile; fileRef = 4225EA57257EC36300923CE3 /* BTPaymentMethodNonce+DropIn.m */; };
-		4225EA87257EC59600923CE3 /* BTPaymentMethodNonce+DropInTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4225EA86257EC59600923CE3 /* BTPaymentMethodNonce+DropInTests.swift */; };
```
Comment (sestevens) [inline BraintreeDropIn.xcodeproj/project.pbxproj:17]: These tests are covered in the base SDK, so we're removing them here.

File: Podfile [hunk 2]
```diff
@@ -1,31 +0,0 @@
-source 'https://cdn.cocoapods.org/'
```
Comment (sestevens) [inline Podfile:1]: We're no longer using CocoaPods for the demo app, so we can remove the Podfile and Podfile.lock.

File: UnitTests/BTPaymentMethodNonce+DropInTests.swift [hunk 3]
```diff
@@ -3,61 +3,40 @@ import BraintreeDropIn
 
 class BTPaymentMethodNonceDropInTests: XCTestCase {
 
-    let sharedParser = BTPaymentMethodNonceParser.shared()
```
Comment (sestevens) [inline UnitTests/BTPaymentMethodNonce+DropInTests.swift:6]: `BTPaymentMethodNonceParser` is now internal to the base SDK, so we can't use it here.

File: UnitTests/BTPaymentMethodNonce+DropInTests.swift [hunk 4]
```diff
@@ -3,61 +3,40 @@ import BraintreeDropIn
...
-        let cardJSON = BTJSON(value: [
-            "nonce": "fake-nonce",
-            "description": "Visa ending in 11",
-            "details": [
-                "lastTwo" : "11",
-                "cardType": "visa"
-            ]
-        ])
-        let cardNonce = sharedParser.parseJSON(cardJSON, withParsingBlockForType: "CreditCard") as? BTCardNonce
+        class MockCardNonce: BTCardNonce {
```
Comment (sestevens) [inline UnitTests/BTPaymentMethodNonce+DropInTests.swift:7]: I created mock classes to work around the fact that the tests need to set properties that are read-only. It's not ideal, but I couldn't come up with a better option.

**Review by scannillo — APPROVED** [review]

**Review by sshropshire — APPROVED** [review]

**Pull request merged**